### PR TITLE
Add windows container build and production setup.

### PR DESCRIPTION
**Pull request by cyriltovena**
This adds build step and production setup to run the agent on windows container with Kubernetes.

Tested it and it works like a charm.

We need to discuss how we should be doing release of this image, I don't have much time to invest so I would hope someone can take over.

### CODE REVIEW

File: .github/workflows/containerize.yml [hunk 1]
```diff
@@ -31,3 +31,16 @@ jobs:
...
+    runs-on: windows-2019
+    steps:
+    - name: Login to Docker Hub
+      run: docker login -u '${{ secrets.DOCKER_USER }}' -p '${{ secrets.DOCKER_PASS }}'
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Build and Push Container 
+      run: |
+        $sha=(git rev-parse --short=7 HEAD)
+        docker build . -f .\cmd\agent\Dockerfile.windows -t grafana/agent:nano-1809-$sha
```
Comment (rfratto) [inline .github/workflows/containerize.yml:45]: Dumb question: is the tag name temporary or does it need to be prefixed with `nano-1809` for some reason?

File: production/kubernetes/agent-loki-windows.yaml [hunk 2]
```diff
@@ -0,0 +1,362 @@
+apiVersion: v1
```
Comment (rfratto) [inline production/kubernetes/agent-loki-windows.yaml:1]: The yaml manifests are generated from Tanka configs so we'll need to set those up to generate this. Could you summarize what you've done here/anything else you've needed to add to get this to work?

Was this file given to any users or do you have it here as an example?

**Review by rfratto — COMMENTED** [review]
Hey @cyriltovena, thanks for working on this! I'll be taking it over and just have some questions so I can move forward.

File: .github/workflows/containerize.yml (hunk 1 above)
Comment (cyriltovena) [reply .github/workflows/containerize.yml:45]: it would be better if we could figure out how to do a multi-arch image e.g windows and linux. I think @shorez might this better than me.

File: production/kubernetes/agent-loki-windows.yaml (hunk 2 above)
Comment (cyriltovena) [reply production/kubernetes/agent-loki-windows.yaml:1]: This is meant to be used in a kubernetes cluster that has windows nodes. Nothing special needs to be done apart from `kubectl apply -f agent-loki-windows.yaml`.

The statefulset will be deployed only on windows node, so if you also have linux nodes you need an other statefulset for that.

File: .github/workflows/containerize.yml (hunk 1 above)
Comment (rfratto) [reply .github/workflows/containerize.yml:45]: I'm using buildx in #161 which will be merged shortly but I think the Dockerfile here is different enough that that wouldn't work for a multiarch build

**Comment by rfratto** [comment]
I'm going to remove the keepalive tag. While I would like to see Windows container support added eventually, it's not a priority and figuring out a way to add it into our build tooling cleanly. We're tracking this in #646 and hoping that the community can pick it up with a proposal.
